File: pnrs-tools-back-java/api/src/main/java/com/pnrstools/api/controllers/NdcController.java
package com.pnrstools.api.controllers;

import com.pnrstools.api.model.dto.CreateNdcRoundTripDto;
import com.pnrstools.api.services.NdcRoundTripService;
import io.swagger.v3.oas.annotations.Operation;
import io.swagger.v3.oas.annotations.tags.Tag;
import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.HashMap;
import java.util.Map;

@RestController
@RequestMapping("/api/ndc")
@RequiredArgsConstructor  // ← Lombok genera constructor con ndcRoundTripService
@Tag(name = "NDC Round-Trip", description = "API para creación de reservas NDC Round-Trip")
public class NdcController {

    private final NdcRoundTripService ndcRoundTripService;  // ← Inyectado por constructor

    @Operation(summary = "Crea una reserva NDC Round-Trip")
    @PostMapping("/round-trip")
    public ResponseEntity<Map<String, Object>> createRoundTrip(@RequestBody CreateNdcRoundTripDto dto) {
        try {
            Map<String, Object> result = ndcRoundTripService.createRoundTripBooking(dto);
            return ResponseEntity.ok(result);
        } catch (Exception e) {
            Map<String, Object> error = new HashMap<>();
            error.put("status", "ERROR");
            error.put("message", e.getMessage());
            return ResponseEntity.status(500).body(error);
        }
    }
    
}

File: pnrs-tools-back/src/controllers/ndc.controller.ts
// src/controllers/ndc.controller.ts

import type { Request, Response } from 'express';
import { NdcRoundTripService } from '../services/ndc-roundtrip.service.js';
import type { CreateNdcRoundTripDto } from '../services/ndc-roundtrip.dto.js';

const ndcService = new NdcRoundTripService();

export class NdcController {
  
  /**
   * @openapi
   * /api/ndc/round-trip:
   *   post:
   *     summary: Crea una reserva NDC Round-Trip
   *     tags:
   *       - NDC Round-Trip
   *     requestBody:
   *       required: true
   *       content:
   *         application/json:
   *           schema:
   *             type: object
   *             properties:
   *               origin:
   *                 type: string
   *                 example: MAD
   *               destination:
   *                 type: string
   *                 example: BRU
   *               outboundDate:
   *                 type: string
   *                 example: 2026-05-10
   *               inboundDate:
   *                 type: string
   *                 example: 2026-05-20
   *               emailContact:
   *                 type: string
   *                 example: test@example.com
   *               phoneContact:
   *                 type: string
   *                 example: 34600000000
   *     responses:
   *       200:
   *         description: Reserva completada con éxito
   *       500:
   *         description: Error en el servicio NDC
   */
  static async createRoundTrip(req: Request, res: Response) {
    try {
      const dto: CreateNdcRoundTripDto = req.body;
      const result = await ndcService.createRoundTripBooking(dto);
      
      return res.status(200).json(result);
    } catch (error: unknown) {
      const errorMessage = error instanceof Error ? error.message : 'Error interno procesando la reserva NDC';
      return res.status(500).json({
        status: 'ERROR',
        message: errorMessage,
      });
    }
  }
  
}
